TelemetryTests/TelemetryTests.swift

```diff
@@ -13,63 +13,70 @@ import OHHTTPStubs
 
 class TelemetryTests: XCTestCase {
     var expectation: XCTestExpectation? = nil
+    var expectedFilesUploaded = 0
+    var countFilesUploaded = 0
 
     override func setUp() {
         super.setUp()
+
+        expectedFilesUploaded = 0
+        countFilesUploaded = 0
+
         // Put setup code here. This method is called before the invocation of each test method in the class.
         stub(condition: isHost("incoming.telemetry.mozilla.org")) { data in
             let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
             let str = String(data: body!, encoding: .utf8) ?? ""
             print(" -- \n \(str)")
-            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            let _ = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
 
-            self.expectation?.fulfill()
+            self.countFilesUploaded += 1
+            if self.expectedFilesUploaded == self.countFilesUploaded {
+                self.expectation?.fulfill()
+            }
             return OHHTTPStubsResponse(jsonObject: ["foo": "bar"], statusCode: 200, headers: ["Content-Type": "application/json"])
         }
     }
-    
+
     override func tearDown() {
         // Put teardown code here. This method is called after the invocation of each test method in the class.
         super.tearDown()
     }
-    
+
     func testTelemetryUpload() {
         let telemetryConfig = Telemetry.default.configuration
         telemetryConfig.appName = "AppInfo.displayName"
         telemetryConfig.userDefaultsSuiteName = "AppInfo.sharedContainerIdentifier"
         telemetryConfig.dataDirectory = .documentDirectory
         Telemetry.default.add(pingBuilderType: CorePingBuilder.self)
 
-        Telemetry.default.recordSessionStart()
-        Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.foreground, object: TelemetryEventObject.app)
-        Telemetry.default.recordSessionEnd()
+        expectedFilesUploaded = 3
 
-        Telemetry.default.queue(pingType: CorePingBuilder.PingType)
-        Telemetry.default.scheduleUpload(pingType: CorePingBuilder.PingType)
+        for _ in 0..<expectedFilesUploaded {
+            Telemetry.default.recordSessionStart()
+            Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.foreground, object: TelemetryEventObject.app)
```

Comment:
If we're only testing `CorePingBuilder` here, this is unnecessary. You can add a `recordSearch()` or two though and it will change the data in the core ping.